messaging/src/test/java/org/axonframework/messaging/timeout/CombinedTimeoutTests.java

```diff
@@ -0,0 +1,241 @@
+package org.axonframework.messaging.timeout;
+
+import org.axonframework.eventhandling.EventMessage;
+import org.axonframework.eventhandling.GenericEventMessage;
+import org.axonframework.messaging.Message;
+import org.axonframework.messaging.annotation.MessageHandlingMember;
+import org.axonframework.messaging.unitofwork.BatchingUnitOfWork;
+import org.junit.jupiter.api.*;
+
+import java.lang.annotation.Annotation;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.concurrent.Callable;
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+
+import static org.awaitility.Awaitility.await;
+import static org.junit.jupiter.api.Assertions.*;
+
+/**
+ * The different timeout components, {@link TimeoutWrappedMessageHandlingMember} and
+ * {@link UnitOfWorkTimeoutInterceptor} are tested in isolation, but this test class combines them to ensure that the
+ * timeout behavior works as expected when both are used together.
+ */
+public class CombinedTimeoutTests {
```

Comment:
```suggestion
class CombinedTimeoutTests {
```

Reply:
Besides this nit, this test case looks comprehensive..nice work!

Reply:
Thanks! :) Making package-private